python/cuml/dask/common/input_utils.py

```diff
@@ -155,6 +144,24 @@ def calculate_parts_to_sizes(self, comms=None, ranks=None):
             self.total_rows += total
 
 
+def get_datatype(data):
```

Comment:
I would consider renaming and/or documenting. Name is very general now. I kind of expected it to return a dtype? Maybe call it something like "get_type_from_inputs'" and have a good docstring? Hmm... I have no good name suggestions...